### PR TITLE
Contibuting personal ErgoDox EZ keymap (davidhbrown)

### PR DESCRIPTION
**Pull request by davidhbrown**
## Description

This keymap tries to be easy to learn and switch to-and-from by maximizing the number of standard keys that are at or reasonably near their standard QWERTY locations while providing a few extras for coding. Additional layers provide function keys, an embedded numeric keypad, and mouse/media controls. The CapsLock, NumLock, and ScrollLock USB Host status states are displayed with the indicator LEDs of the keyboard which do double-duty to also indicate activation of the three additional layers.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* (none)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).
Both `make ergodox_ez:davidhbrown` and `make ergodox_ez:default` work without emitting errors or warnings under MSYS2 MinGW and QMK Firmware 0.6.453.

### CODE REVIEW

File: keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+rgblight_config_t rgblight_config;
+bool disable_layer_color = 0;
+
+bool suspended = false;
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  if (record->event.pressed) {
+    switch (keycode) {
+      case EPRM:
```
Comment (noroadsleft) [inline keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c]: QMK has a keycode to do this built-in. You can use `EEP_RST` in place of `EPRM`.

https://github.com/qmk/qmk_firmware/blob/92f9b6c3bdff939bc562d640e02c3aebaa204e17/quantum/quantum.c#L323-L327
https://github.com/qmk/qmk_firmware/blob/92f9b6c3bdff939bc562d640e02c3aebaa204e17/quantum/quantum_keycodes.h#L613

File: keyboards/ergodox_ez/keymaps/davidhbrown/readme.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+I pulled out the code to show both USB Host status LEDs and layer switches into a separate file; notes on how to add that to your own keymap are described later in the section about `statusLed.c`, below 
+
+# Keymap extended description
+I found this arrangement easier to learn and adjust to than the default layout, especially because I often switch to any of a number of laptop keyboards when I'm not at my desktop machine.
+
+If you would like to download a hex file with this layout including the statusLed.c behavior, you can get that from https://davidhbrown.us/ergodox_ez/ergodox_ez_davidhbrown.hex  This works with both my ErgoDox EZ Original and my ErgoDox EZ Shine keyboards.
+
+If you would like to fork the layout in the configurator, go to [https://configure.ergodox-ez.com/ergodox-ez/layouts/LvADM/latest/0](https://configure.ergodox-ez.com/ergodox-ez/layouts/LvADM/latest/0). What you download won't have the statusLed function; that's not part of the configurator. The configrator is what prepared the nice layout images.
```
Comment (noroadsleft) [inline keyboards/ergodox_ez/keymaps/davidhbrown/readme.md]: Typo.

```suggestion
If you would like to fork the layout in the configurator, go to [https://configure.ergodox-ez.com/ergodox-ez/layouts/LvADM/latest/0](https://configure.ergodox-ez.com/ergodox-ez/layouts/LvADM/latest/0). What you download won't have the statusLed function; that's not part of the configurator. The configurator is what prepared the nice layout images.
```

File: .vscode/settings.json [hunk 3]
```diff
@@ -13,5 +13,6 @@
         "*.cpp": "cpp",
         "*.hpp": "cpp",
         "xstddef": "c"
-    }
+    },
```
[the remark this answers is not in the document]
Comment (drashna) [reply .vscode/settings.json]: I think we really need to add the entire .vscode folder to gitignore. VS Code is adding more and more stuff that gets stored there....

But for now, yeah, this should be reverted.

File: keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c (hunk 1 above)
Comment (drashna) [reply keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c]: @noroadsleft this is from the default keymap, and needs to be updated (in fact, my ZSA PR updates this)

File: .vscode/settings.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (davidhbrown) [reply .vscode/settings.json]: Whoops. I hadn't intended any file outside of the davidhbrown folder to be part of this. (I don't usually use VS Code except that GitHub Desktop started opening that by default.) I agree that it should probably be in .gitignore. I have changed my settings.json back to what's in master and will commit again...

File: keyboards/ergodox_ez/keymaps/davidhbrown/readme.md (hunk 2 above)
Comment (davidhbrown) [reply keyboards/ergodox_ez/keymaps/davidhbrown/readme.md]: Thank you for spotting that :-)

File: keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+enum custom_keycodes {
+  EPRM = SAFE_RANGE,
+  VRSN,
+  RGB_SLD
+};
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_ergodox(KC_ESCAPE,KC_1,KC_2,KC_3,KC_4,KC_5,KC_6,KC_TAB,KC_Q,KC_W,KC_E,KC_R,KC_T,KC_CALCULATOR,KC_LCBR,KC_A,KC_S,KC_D,KC_F,KC_G,KC_LSPO,KC_Z,KC_X,KC_C,KC_V,KC_B,KC_ENTER,KC_LCTRL,MO(1),KC_LGUI,KC_LALT,LT(3,KC_GRAVE),KC_PSCREEN,KC_APPLICATION,KC_HOME,KC_SPACE,KC_BSPACE,KC_END,KC_6,KC_7,KC_8,KC_9,KC_0,KC_MINUS,KC_EQUAL,KC_TAB,KC_Y,KC_U,KC_I,KC_O,KC_P,KC_BSLASH,KC_H,KC_J,KC_K,KC_L,KC_QUOTE,KC_ENTER,KC_NUMLOCK,KC_N,KC_M,KC_COMMA,KC_DOT,KC_UP,KC_RSPC,RALT_T(KC_SCOLON),RCTL_T(KC_SLASH),KC_LEFT,KC_DOWN,KC_RIGHT,KC_LBRACKET,KC_RBRACKET,KC_PGUP,KC_PGDOWN,KC_DELETE,KC_SPACE),
```
[the remark this answers is not in the document]
Comment (davidhbrown) [reply keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c]: Yeah... I just copied over what the ErgoDox configurator gave me, and it's certainly possible that that service would not be available in the future. I don't suppose there's a script somewhere that prettifies these lists? Is this important enough to be blocking? Sort of feels like a YAGNI, but I'm willing to reformat it.

File: keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c (hunk 1 above)
Comment (davidhbrown) [reply keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c]: I'm not sure what (if anything) I should be doing in response to these comments. Does it help that I don't use macro recording in the keymap? I think I could probably remove this function. (Yes, I did start with the default keymap.c plus the code arrays from the configurator).

File: keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c [hunk 5]
```diff
@@ -0,0 +1,99 @@
+#include QMK_KEYBOARD_H
+#include "version.h"
+
+#define BASE 0 // default layer
```
[the remark this answers is not in the document]
Comment (davidhbrown) [reply keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c]: Oh, nice. Thank you.

File: keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (noroadsleft) [reply keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c]: If you want to leave it as-is that's fine. It was simply a suggestion.

File: keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c (hunk 1 above)
Comment (drashna) [reply keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c]: Yeah, it should be removed.

File: keyboards/ergodox_ez/keymaps/davidhbrown/config.h [hunk 6]
```diff
@@ -0,0 +1,8 @@
+/*
+  Set any config.h overrides for your specific keymap here.
+  See config.h options at https://docs.qmk.fm/#/config_options?id=the-configh-file
+*/
+#undef RGBLIGHT_HUE_STEP
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/davidhbrown/config.h]:
```suggestion
#pragma once 

#undef RGBLIGHT_HUE_STEP
```

Comment (davidhbrown) [reply keyboards/ergodox_ez/keymaps/davidhbrown/config.h]: (Thanks for the poke; I got busy with the beginning of the semester. I will try to do this this weekend.)

**Comment by davidhbrown** [comment]
As you can see, I had some trouble getting my build environment updated to 0.7.25. I'm concerned that the issue I had with the lufa submodule is now affecting this merge request. I'm not perfectly sure how  to fix that...  maybe I can roll back to 6d4a829 or even one of the earlier ones...

**Comment by drashna** [comment]
For the LUFA thing,  you may just need to run `make git-submodule` to fix this.

If that doesn't work, go to the folder (eg run `cd lib/lufa`) and run `git reset ce10f7642b0459e409839b23cc91498945119b4d --hard` . And that should fix the issue.

**Pull request closed without merging**